code/game/objects/items/weapons/batons.dm

```diff
@@ -0,0 +1,127 @@
+/// Delay in deci-seconds between two non-lethal attacks
+#define BATON_STUN_COOLDOWN 4 SECONDS
+/// Force of the telescopic baton when deployed
+#define BATON_TELESCOPIC_FORCE_DEPLOYED 10
+
+/**
+  * # Police Baton
+  *
+  * Knocks down the hit mob when not on harm intent and when [/obj/item/melee/classic_baton/on] is TRUE
+  *
+  * A non-lethal attack has a cooldown to avoid spamming
+  */
+/obj/item/melee/classic_baton
+	name = "police baton"
+	desc = "A wooden truncheon for beating criminal scum."
+	icon_state = "baton"
+	item_state = "classic_baton"
+	slot_flags = SLOT_BELT
+	force = 12 //9 hit crit
+	w_class = WEIGHT_CLASS_NORMAL
+	/// Whether the baton is on cooldown
+	var/on_cooldown = FALSE
+	/// Whether the baton is toggled on (to allow attacking)
+	var/on = TRUE
+
+/obj/item/melee/classic_baton/attack(mob/target, mob/living/user)
+	if(!on)
+		return ..()
+
+	add_fingerprint(user)
+	if((CLUMSY in user.mutations) && prob(50))
+		user.visible_message("<span class='danger'>[user] accidentally clubs [user.p_them()]self with [src]!</span>", \
+							"<span class='userdanger'>You accidentally club yourself with [src]!</span>")
+		user.Weaken(force * 3)
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			H.apply_damage(force * 2, BRUTE, "head")
+		else
+			user.take_organ_damage(force * 2)
+		return
+	if(isrobot(target))
+		return ..()
+	if(!isliving(target))
+		return
+
+	if(user.a_intent == INTENT_HARM) // Lethal attack
+		if(!..() || !isrobot(target))
```

Comment:
isrobot can never be true here.
This is best to be rewritten as `return ..()`
How it's written currently is a bit odd